File: PST/FUEL Source Code/FUEL/WindowWrapper.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.WindowWrapper
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using System;
using System.Diagnostics;
using System.Windows.Forms;

namespace FUEL
{
  internal class WindowWrapper : IWin32Window
  {
    private IntPtr _hwnd { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    internal WindowWrapper(IntPtr aHandle) => this._hwnd = aHandle;

    IntPtr IWin32Window.Handle => this._hwnd;

    internal IntPtr WindowWrapper
    {
      set => this._hwnd = value;
    }
  }
}

File: PST/FUEL/src/FUEL/WindowWrapper.vb
Imports System
Imports System.Diagnostics
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    Friend Class WindowWrapper
        Implements IWin32Window
        ' Methods
        Friend Sub New(ByVal aHandle As IntPtr)
            Me._hwnd = aHandle
        End Sub


        ' Properties
        Private Property _hwnd As IntPtr
            <DebuggerNonUserCode> _
            Get
                Return Me.__hwnd
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As IntPtr)
                Me.__hwnd = AutoPropertyValue
            End Set
        End Property

        Friend ReadOnly Property Handle As IntPtr
            Get
                Return Me._hwnd
            End Get
        End Property

        Friend WriteOnly Property WindowWrapper As IntPtr
            Set(ByVal value As IntPtr)
                Me._hwnd = value
            End Set
        End Property

        Friend ReadOnly Property System.Windows.Forms.IWin32Window.Handle As IntPtr
            Get
                Return Me._hwnd
            End Get
        End Property


        ' Fields
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __hwnd As IntPtr
    End Class
End Namespace

